### PR TITLE
Feat/1201965382322491 did rpc

### PR DESCRIPTION
**Pull request by sfffaaa**
1. Add the RPC/runtime RPC module
2. Restructure the file hierarchy

### CODE REVIEW

File: rpc/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            validity: item.validity,
+            created: item.created,
+        }
+    }
+}
+
+#[rpc]
+pub trait PeaqDIDApi<BlockHash, AccountId, BlockNumber, Moment> {
+	#[rpc(name = "peaqdid_readAttributes")]
+	fn read_attributes(&self, did_account: AccountId, name: Bytes, at: Option<BlockHash>) ->
```
Comment (irediaes) [inline rpc/src/lib.rs]: I will suggest the function name should be `read_attribute` because a single attribute is returned

File: rpc/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+/// A struct that implements the [`PeaqDIDApi`].
+pub struct PeaqDID<C, B> {
+	client: Arc<C>,
+	_marker: std::marker::PhantomData<B>,
+}
+
+impl<C, B> PeaqDID<C, B> {
+	/// Create new `Oracle` with the given reference to the client.
```
Comment (irediaes) [inline rpc/src/lib.rs]: What do you mean by "oracle" on the comment? Or is it a typo?

File: rpc/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+impl<C, Block, AccountId, BlockNumber, Moment> PeaqDIDApi<<Block as BlockT>::Hash, AccountId, BlockNumber, Moment> for PeaqDID<C, Block>
+where
+	Block: BlockT,
+	C: Send + Sync + 'static + ProvideRuntimeApi<Block> + HeaderBackend<Block>,
+	C::Api: PeaqDIDRuntimeApi<Block, AccountId, BlockNumber, Moment>,
+	AccountId: Codec,
+	BlockNumber: Codec,
+	Moment: Codec,
+{
+	fn read_attributes(&self, did_account: AccountId, name: Bytes, at: Option<<Block as BlockT>::Hash>) ->
```
Comment (irediaes) [inline rpc/src/lib.rs]: I will suggest the function name should be `read_attribute` because a single attribute is returned

File: rpc/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            value: item.value.into(),
+            validity: item.validity,
+            created: item.created,
+        }
+    }
+}
+
+#[rpc]
+pub trait PeaqDIDApi<BlockHash, AccountId, BlockNumber, Moment> {
+	#[rpc(name = "peaqdid_readAttributes")]
```
Comment (irediaes) [inline rpc/src/lib.rs]: Same with the rpc name `peaqdid_readAttribute`

File: rpc/src/lib.rs (hunk 2 above)
Comment (sfffaaa) [reply rpc/src/lib.rs]: Typo, thanks a lot!

File: rpc/src/lib.rs (hunk 1 above)
Comment (sfffaaa) [reply rpc/src/lib.rs]: Updated

File: rpc/src/lib.rs (hunk 3 above)
Comment (sfffaaa) [reply rpc/src/lib.rs]: Updated

File: rpc/src/lib.rs (hunk 4 above)
Comment (sfffaaa) [reply rpc/src/lib.rs]: Updated

**Review by irediaes — APPROVED** [review]

**Pull request merged**